### PR TITLE
Show page chart

### PR DESCRIPTION
**Pull request by nbondarenko**

### CODE REVIEW

File: et [hunk 1]
```diff
@@ -0,0 +1,289 @@
+=> {[31m[1;31m"[0m[31m2016-10-03 11:50:12[1;31m"[0m[31m[0m=>[1;34m0[0m,
```
Comment (tilast) [inline et]: do we need this file at all?

File: frontend/js/response_chart.js [hunk 2]
```diff
@@ -1,64 +1,41 @@
...
+    var parentContainer = $('#checks_chart');
+    parentContainer.html("");
+    var width = parentContainer.width();
+    var height = parentContainer.height();
+    var logOfTimes = _.keysIn(data.log);
+    var log = _.toArray(data.log);
+    var logOfFails = _.toArray(data.fails_log);
+    var logOfIssues = _.toArray(data.issues_log);
+    var amount = log.length;
+    var max = Math.max.apply(Math, log);
```
Comment (tilast) [inline frontend/js/response_chart.js]:
``` js
  var max = _.max(log) || _.max(logOfFails);
```

File: frontend/js/response_chart.js [hunk 3]
```diff
@@ -1,64 +1,41 @@
...
+      parentContainer.append("<div class= 'chart-bar' data-id="+ i +"></div>");
+      if(log[i])
+        $("div[data-id="+ i +"]").width(100/amount+"%").height(299/max*log[i]).addClass("hint--bottom hint--small").attr("aria-label", logOfTimes[i]);
+      else if(logOfFails[i])
+        $("div[data-id="+ i +"]").width(100/amount+"%").height(299/max*logOfFails[i]).addClass("fail hint--bottom hint--small hint--warning").attr("aria-label",  logOfTimes[i] + " - " + logOfIssues[i]);
+      else
+        $("div[data-id="+ i +"]").width(100/amount+"%").height(299/max*logOfFails[i]).addClass("network-fail hint--bottom hint--small hint--error").attr("aria-label",  logOfTimes[i] + " - " + logOfIssues[i]);
+      if(i%20 == 0){
+        $("div[data-id="+ i +"]").append("<span>" + logOfTimes[i].slice(11) + "</span>");
       }
```
Comment (tilast) [inline frontend/js/response_chart.js:31]: plz omit any constructions without the `{}`, like for `if` or `else`

File: frontend/js/response_chart.js [hunk 4]
```diff
@@ -1,64 +1,41 @@
...
+    var logOfFails = _.toArray(data.fails_log);
+    var logOfIssues = _.toArray(data.issues_log);
+    var amount = log.length;
+    var max = Math.max.apply(Math, log);
+    if(!max)
+      max = Math.max.apply(Math, logOfFails);
+    for(var i = 0; i < amount; i++){
+      parentContainer.append("<div class= 'chart-bar' data-id="+ i +"></div>");
+      if(log[i])
+        $("div[data-id="+ i +"]").width(100/amount+"%").height(299/max*log[i]).addClass("hint--bottom hint--small").attr("aria-label", logOfTimes[i]);
```
Comment (tilast) [inline frontend/js/response_chart.js]: `100`, `299` - magic numbers?

Comment (tilast) [inline frontend/js/response_chart.js]: plz prefer using `'` over the `"`

File: frontend/js/response_chart.js [hunk 5]
```diff
@@ -1,64 +1,41 @@
...
+      else if(logOfFails[i])
+        $("div[data-id="+ i +"]").width(100/amount+"%").height(299/max*logOfFails[i]).addClass("fail hint--bottom hint--small hint--warning").attr("aria-label",  logOfTimes[i] + " - " + logOfIssues[i]);
+      else
+        $("div[data-id="+ i +"]").width(100/amount+"%").height(299/max*logOfFails[i]).addClass("network-fail hint--bottom hint--small hint--error").attr("aria-label",  logOfTimes[i] + " - " + logOfIssues[i]);
+      if(i%20 == 0){
+        $("div[data-id="+ i +"]").append("<span>" + logOfTimes[i].slice(11) + "</span>");
```
Comment (tilast) [inline frontend/js/response_chart.js]: `11` - magic number?

File: frontend/js/response_chart.js [hunk 6]
```diff
@@ -1,64 +1,41 @@
...
+      max = Math.max.apply(Math, logOfFails);
+    for(var i = 0; i < amount; i++){
+      parentContainer.append("<div class= 'chart-bar' data-id="+ i +"></div>");
+      if(log[i])
+        $("div[data-id="+ i +"]").width(100/amount+"%").height(299/max*log[i]).addClass("hint--bottom hint--small").attr("aria-label", logOfTimes[i]);
+      else if(logOfFails[i])
+        $("div[data-id="+ i +"]").width(100/amount+"%").height(299/max*logOfFails[i]).addClass("fail hint--bottom hint--small hint--warning").attr("aria-label",  logOfTimes[i] + " - " + logOfIssues[i]);
+      else
+        $("div[data-id="+ i +"]").width(100/amount+"%").height(299/max*logOfFails[i]).addClass("network-fail hint--bottom hint--small hint--error").attr("aria-label",  logOfTimes[i] + " - " + logOfIssues[i]);
+      if(i%20 == 0){
```
Comment (tilast) [inline frontend/js/response_chart.js]: `20` - magic number?

File: gulpfile.js [hunk 7]
```diff
@@ -26,6 +27,8 @@ var paths = {
     'bower_components/parsleyjs/src/parsley.css',
     'bower_components/c3/c3.min.css',
     'bower_components/select2/dist/css/select2.min.css',
+    'bower_components/hint.css/hint.min.css',
+    // 'bower_components/tooltipster/dist/css/tooltipster.bundle.min.css',
```
Comment (tilast) [inline gulpfile.js]: plz remove or uncomment commented code

**Review by tilast — CHANGES_REQUESTED** [review]

File: app/controllers/checks_controller.rb [hunk 8]
```diff
@@ -44,7 +44,8 @@
   results = check.results.order(:created_at.asc)
   log = prepare_logs(results, is_ok: true)
   fails_log = prepare_logs(results, is_ok: false)
-  json({ log: log, fails_log: fails_log })
+  issues_log = get_issues(results);
+  json({ log: log, fails_log: fails_log, issues_log: issues_log })
```
Comment (denysxftr) [inline app/controllers/checks_controller.rb]: you can remove useless code here
plz use only `log` var which has structure like 

``` ruby
[
  { time: '12:30:21', issues: 'joined issues here', response_time: 456 },
  { time: '12:35:15', issues: 'joined issues here', response_time: 317 }
]
```

and make appropriate changes in js file

**Review by denysxftr — CHANGES_REQUESTED** [review]
plz rework a bit data structure which you pass from controller and make related changes in js

File: frontend/js/response_chart.js [hunk 9]
```diff
@@ -3,31 +3,42 @@ $(function(){
...
-      if(log[i])
-        $("div[data-id="+ i +"]").width(100/amount+"%").height(299/max*log[i]).addClass("hint--bottom hint--small").attr("aria-label", logOfTimes[i]);
-      else if(logOfFails[i])
-        $("div[data-id="+ i +"]").width(100/amount+"%").height(299/max*logOfFails[i]).addClass("fail hint--bottom hint--small hint--warning").attr("aria-label",  logOfTimes[i] + " - " + logOfIssues[i]);
-      else
-        $("div[data-id="+ i +"]").width(100/amount+"%").height(299/max*logOfFails[i]).addClass("network-fail hint--bottom hint--small hint--error").attr("aria-label",  logOfTimes[i] + " - " + logOfIssues[i]);
-      if(i%20 == 0){
-        $("div[data-id="+ i +"]").append("<span>" + logOfTimes[i].slice(11) + "</span>");
+      parentContainer.append('<div class= "chart-bar" data-id='+ i +'></div>');
+      if(log[i]){
```
Comment (tilast) [inline frontend/js/response_chart.js]: plz read https://github.com/airbnb/javascript

File: frontend/js/response_chart.js [hunk 10]
```diff
@@ -1,64 +1,51 @@
...
+    var parentContainer = $('#checks_chart');
+    parentContainer.html('');
+    var width = 100;
+    var legendScale = 20;
+    var height = parentContainer.height();
+    var log = data.log;
+    var amount = log.length;
+    var maxOfTimeouts = _.maxBy(log, function(item){ return item.timeout; }).timeout;
+    log.forEach(function(item, index){
+      parentContainer.append('<div class= "chart-bar" data-id='+ index +'></div>');
```
Comment (denysxftr) [inline frontend/js/response_chart.js]:
``` js
var bar = $('<div class="chart-bar" data-id=' + index + '></div>');
```

**Review by denysxftr — CHANGES_REQUESTED** [review]

**Review by tilast — APPROVED** [review]
ok for me, but let's wait for @ololyay

File: app/controllers/checks_controller.rb [hunk 11]
```diff
@@ -42,9 +42,7 @@
   protect!
   check = Check.find(params[:id])
   results = check.results.order(:created_at.asc)
-  log = prepare_logs(results, is_ok: true)
-  fails_log = prepare_logs(results, is_ok: false)
-  json({ log: log, fails_log: fails_log })
+  json({ log: prepare_logs(results)})
```
Comment (dvoitekh) [inline app/controllers/checks_controller.rb:45]: add space before `}` please

**Review by dvoitekh — APPROVED** [review]

File: frontend/js/response_chart.js [hunk 12]
```diff
@@ -1,64 +1,49 @@
...
+
+  function drawDiagram(data) {
+    var parentContainer = $('#checks_chart');
+    parentContainer.html('');
+    var width = 100;
+    var legendScale = 20;
+    var height = parentContainer.height();
+    var log = data.log;
+    var amount = log.length;
+    var maxOfTimeouts = _.maxBy(log, item =>  { return item.timeout; } ).timeout;
```
Comment (dvoitekh) [inline frontend/js/response_chart.js]: too many spaces here (remove one after `=>` and one after `}`)

**Review by denysxftr — APPROVED** [review]
nice
plz fix @dvoitekh comments

File: frontend/js/response_chart.js (hunk 12 above)
Comment (dvoitekh) [reply frontend/js/response_chart.js]: please try using 

``` es6
var maxOfTimeouts = _.maxBy(log, item => item.timeout).timeout;
```

Comment (tilast) [reply frontend/js/response_chart.js]: do we have a es6 here? @ololyay

Comment (denysxftr) [reply frontend/js/response_chart.js]: nope

Comment (tilast) [reply frontend/js/response_chart.js]: well then it will not work at all, should be a syntax error

Comment (dvoitekh) [reply frontend/js/response_chart.js]: ¯_(ツ)_/¯

Comment (nbondarenko) [reply frontend/js/response_chart.js]: there are no errors, but I'll check

Comment (tilast) [reply frontend/js/response_chart.js]: well there is no error in chrome because it supports es6 natively
but I'd not rely on that

Comment (nbondarenko) [reply frontend/js/response_chart.js]: works in firefox too

Comment (tilast) [reply frontend/js/response_chart.js]: I know, one of the _latest_ firefoxes
but the browsers support is still poor
https://kangax.github.io/compat-table/es6/

**Review by tilast — APPROVED** [review]

**Pull request merged**